Userland/Services/WebContent/InputSource.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include "AK/Debug.h"
+#include <AK/HashTable.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/String.h>
+
+namespace Web::WebDriver {
+
+// InputState has lists of InputSources, but the pointer input source constructor
+// depends on InputState, so the definitions are circular
+// forward declare InputState here
+class InputState;
+
+// https : // w3c.github.io/webdriver/#input-sources
+enum class InputSourceType { none,
+    key,
+    pointer,
+    wheel };
+// this enum is for the types of devices, which make up the subtype field of the PointerInputSource class
+enum class PointingDeviceSubtype { mouse,
+    pen,
+    touch };
+
+class InputSource {
+    // Each input source has an input id which is stored as a key in the input state map.
+    // input id is a UUID
+    AK::String input_id;
```

Comment:
- `m_input_id`, and we don't use the `AK::` prefix for types mostly.
- Should appear after the constructor.
- You'll probably want a public getter for this. eg,
```c++
    String const& input_id() const { return m_input_id; }
```